pymongo/encryption.py

```diff
@@ -552,6 +554,95 @@ def __init__(
+          - `kms_provider`: the KMS provider to be used
+          - `data_key_opts` (dict): a dictionary containing additional arguments to the `create_data_key` method that will be
+            used to auto-generate the data keys if `keyId` is `None` such as::
+
+                {
+                   masterKey: Optional<Document>
+                   keyAltNames: Optional<Array[String]>
+                   keyMaterial: Optional<BinData>
+                }
+
+          - ``encryptedFields`` (optional) (dict): **(BETA)** Document that describes the encrypted fields for
+            Queryable Encryption. For example::
+
+                {
+                  "escCollection": "enxcol_.encryptedCollection.esc",
+                  "eccCollection": "enxcol_.encryptedCollection.ecc",
+                  "ecocCollection": "enxcol_.encryptedCollection.ecoc",
+                  "fields": [
+                      {
+                          "path": "firstName",
+                          "keyId": Binary.from_uuid(UUID('00000000-0000-0000-0000-000000000000')),
+                          "bsonType": "string",
+                          "queries": {"queryType": "equality"}
+                      },
+                      {
+                          "path": "ssn",
+                          "keyId": Binary.from_uuid(UUID('04104104-1041-0410-4104-104104104104')),
+                          "bsonType": "string"
+                      }
+                    ]
+                }
+
+            The "keyId" may be set to `None` to auto-generate the data keys.
+          - `**kwargs` (optional): additional keyword arguments are the same as "create_collection".
+
+        All optional `create collection command`_ parameters should be passed
+        as keyword arguments to this method.
+        See the documentation for :meth:`~pymongo.database.Database.create_collection` for all valid options.
+
+        .. versionadded:: 4.4
+           Added the codec_options, read_preference, and write_concern options.
+
+        .. _create collection command:
+            https://mongodb.com/docs/manual/reference/command/create
+
+        """
+        if not data_key_opts:
+            data_key_opts = {}
+        encrypted_fields = database._get_encrypted_fields(kwargs, name, False)
+        if encrypted_fields:
+            for field in encrypted_fields["fields"]:
+                if isinstance(field, dict) and field.get("keyId") is None:
+                    try:
+                        field["keyId"] = self.create_data_key(
+                            kms_provider=kms_provider,  # type:ignore[arg-type]
+                            **data_key_opts,
+                        )
+                    except EncryptionError as exc:
+                        raise EncryptionError(
+                            Exception(
+                                f"Error occurred while creating data key for field {field['path']} with encryptedFields={str(encrypted_fields)}"
```

Comment:
I'd recommend using repr-style formatting for path and encrypted_fields here.

Reply:
Done.